SPDX-FileCopyrightText: 2021 Melg Eight <public.melg8@gmail.com>

SPDX-License-Identifier: GPL-3.0-or-later

Remove dependency on unicodeio.h. This drops support for outputting unicode
characters. We do that, because inclusion of unicodeio.h from gnulib brings
dependencies on gperf generated files. We don't have gperf on this stage so
we can't regenerate them.

--- ./src/printf.c
+++ ./src/printf.c
@@ -59,7 +59,6 @@
 #include "die.h"
 #include "error.h"
 #include "quote.h"
-#include "unicodeio.h"
 #include "xprintf.h"
 
 /* The official name of this program (e.g., no 'g' prefix).  */
@@ -292,8 +291,12 @@ print_esc (const char *escstart, bool octal_0)
           || (uni_value >= 0xd800 && uni_value <= 0xdfff))
         die (EXIT_FAILURE, 0, _("invalid universal character name \\%c%0*x"),
              esc_char, (esc_char == 'u' ? 4 : 8), uni_value);
-
-      print_unicode_char (stdout, uni_value, 0);
+       putchar ('\\');                                                           
+       if (*p)                                                                   
+         {                                                                       
+           putchar (*p);                                                         
+           p++;                                                                  
+         }  
     }
   else
     {
